src/libmongoc/src/mongoc/mongoc-topology-description-private.h

```diff
@@ -53,16 +53,16 @@ struct _mongoc_topology_description_t {
    bool stale;
    unsigned int rand_seed;
 
+   /* Has this been 'opened' by a log_and_monitor instance? If yes, this is that instance's version_id. Zero if not. */
+   bson_oid_t opened_by_log_and_monitor_version_id;
```

Comment:
> Early SDAM events may be missed due to the relatively late stage at which callbacks can be attached to a non-pooled client.

Is this needed only for tests? I expect this is a change in events returned:

```c
mongoc_client_t *client = mongoc_client_new ("mongodb://localhost:27017");
ASSERT (mongoc_client_command_simple (client, "db", tmp_bson ("{'hello': 1}"), NULL, NULL, NULL));
set_callbacks (client); // Set callbacks after first command.
ASSERT (mongoc_client_command_simple (client, "db", tmp_bson ("{'ping': 1}"), NULL, NULL, NULL));
```

Before this PR, the topology opening callback is not called. If this change is only motivated by tests, can the test runner be modified to set callbacks earlier? That might avoid an unexpected subtle behavior change.

Reply:
Resolving. Latest changes switch back to an `opened` bool flag.